source-academy/frontend
# Restore original functionality to run-all-testcases button

### Description

Restore the original behaviour of the run button when the autograder tab is active, which is to run the program followed by all testcases.

Also switch to `takeLatest` instead of `takeEvery` to avoid race conditions if two runs are executed concurrently.

### Type of change


### How to test

Click on "Run" while the autograder tab is active on a question which has no testcases.

Program should still run as usual.

### Checklist
